docs/reference/mapping/date-format.asciidoc

```diff
@@ -211,6 +211,17 @@ date formatter in that case.
 |=======================================================================
 
 [float]
+[[strict]]
+=== Strict date formats
```

Comment:
I think we should make it more explicit than just a comment a the end of the section. We want to encourage people to use this variant when possible. Maybe we should add this explanation to the beginning of the section and have the patterns read `[strict_]year_month` (or something similar)